e2e/wdio/core/tests/step-input.e2e-spec.ts

```diff
@@ -0,0 +1,159 @@
+import { StepInputPo } from '../pages/step-input.po';
+import {
+    refreshPage,
+    getElementArrayLength,
+    click,
+    getValue,
+    doubleClick,
+    pause,
+    getText,
+    clearValue,
+    setValue,
+    getElementClass,
+    sendKeys,
+    isElementDisplayed
+} from '../../driver/wdio';
+import { sections } from '../fixtures/appData/step-input-content'
+
+describe('Time component test', function () {
+    const stepInputPage = new StepInputPo();
+    const {
+        formExample,
+        labelExample,
+        localExample,
+        stateExample,
+        configExample,
+        defaultExample,
+        currencyExample,
+        step,
+        input,
+        plusButton,
+        minusButton,
+        text,
+        textForDisabledExample
+    } = stepInputPage;
+
+    beforeAll(() => {
+        stepInputPage.open();
+    }, 1);
+
+    afterEach(() => {
+        refreshPage();
+    }, 2);
+
+    it('should check default example inputs', () => {
+        for (let i = 0; i < sections.length; i++) {
+            checkInputWorking(sections[i], 'click', 'positive')
+            checkInputWorking(sections[i], 'click', 'negative')
+            checkInputWorking(sections[i], 'type', 'negative')
+            checkInputWorking(sections[i], 'type', 'positive')
+        }
+    })
+
+    it('should check limitation for step-input', () => {
+
+        setValue(configExample + input, '10', 5);
+        expect(getElementClass(configExample + plusButton, 5)).toContain('is-disabled')
+        setValue(configExample + input, '-10', 5);
+        expect(getElementClass(configExample + minusButton, 5)).toContain('is-disabled')
+
+    })
+
+    it('should verify that in specific input step is 0.5', () => {
+        setValue(configExample + plusButton, '0', 6);
+        for (let i = 0; i < 6; i++) {
+            click(configExample + plusButton, 6)
+        }
+        expect(parseFloat(getValue(configExample + input, 6)) / 6).toEqual(0.5)
+    })
+
+    it('should check Saudi Arabia locale', () => {
+        setValue(localExample + input, '5', 2);
+        sendKeys('Enter')
+        expect(getValue(localExample + input, 2)).toEqual('٥')
+    })
+
+    it('should check input status for Input States example', () => {
+        expect(getElementClass(stateExample + step, 0)).toContain('is-information');
+        expect(getElementClass(stateExample + step, 1)).toContain('is-success');
+        expect(getElementClass(stateExample + step, 2)).toContain('is-warning');
+        expect(getElementClass(stateExample + step, 3)).toContain('is-error');
+    })
+
+    it('should check entering invalid values in inputs', () => {
+        for (let i = 0; i < sections.length; i++) {
+            checkInputWithInvalidValues(sections[i])
+        }
+    })
+
+    it('should check disabled inputs', () => {
+        const defaultValue = getValue(formExample + input, 2);
```

Comment:
instead, can get element by disabled class and check that it's buttons and input aren't clickable
for (...) {
expect(isClickable(selector, index)).toBe(false);
}
or
expect(checkElArrIsClickable(selector)).toBe(false);